Source/Core/DiscIO/VolumeVerifier.cpp

```diff
@@ -541,9 +546,26 @@ bool VolumeVerifier::CheckPartition(const Partition& partition)
     return false;
   }
 
+  // Prepare for hash verification in the Process step
+  if (m_volume.SupportsIntegrityCheck())
+  {
+    u64 offset = m_volume.PartitionOffsetToRawOffset(0, partition);
+    const std::optional<u64> size =
+        m_volume.ReadSwappedAndShifted(partition.offset + 0x2bc, PARTITION_NONE);
+    const u64 end_offset = offset + size.value_or(0);
+
+    for (size_t i = 0; offset < end_offset; ++i, offset += VolumeWii::BLOCK_TOTAL_SIZE)
+      m_blocks.emplace_back(BlockToVerify{partition, offset, i});
+
+    m_block_errors.emplace(partition, 0);
+  }
+
   const DiscIO::FileSystem* filesystem = m_volume.GetFileSystem(partition);
   if (!filesystem)
   {
+    if (m_is_datel)
+      return true;  // Datel Wii Freeloader has an invalid FST in its only partition
```

Comment:
Comments should generally be placed above the respective line instead of off to the side for ease of refactoring.